### PR TITLE
Raw string error note

### LINKED ISSUE #48395: unterminated raw string error should explain the raw-string format a bit
[Playground](https://play.rust-lang.org/?gist=26ef35bce10415e1b6a1cbb61bf9ee97&version=undefined):

```rust
fn foo(s: &str) { println!("{}", s) }
fn main() {
    foo(r#"
This is a raw string  
missing a # at the end
"
    );
}
```

produces

```shell
error: unterminated raw string
 --> src/main.rs:4:9
  |
4 |       foo(r#"
  |  _________^
5 | | This is a raw string  
6 | | missing a # at the end
7 | | "
8 | |     );
9 | | }
  | |_^

error: aborting due to previous error
```

that only mentions "unterminated raw-string". It would be more helpful if it would suggest adding a `#` after the last `"` to terminate the raw string.

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #48395.

I think this note should be helpful enough to solve the issue.

### CODE REVIEW

File: src/test/ui/raw_string.rs [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+fn main() {
+  let x = r##"lol"#;
+  //~^ ERROR unterminated raw string
+}
```
Comment (varkor) [inline src/test/ui/raw_string.rs]: Nit: should be newline-terminated.

Comment (GuillaumeGomez) [reply src/test/ui/raw_string.rs]: Thanks! Fixed.

File: src/test/ui/raw_string.stderr [hunk 2]
```diff
@@ -0,0 +1,11 @@
+error: unterminated raw string
+  --> $DIR/raw_string.rs:12:13
+   |
+12 |       let x = r##"lol"#;
+   |  _____________^
+13 | |     //~^ ERROR unterminated raw string
+14 | | }
+   | |__^
+   |
+   = note: this raw string should be terminated with `"##`
```
Comment (estebank) [inline src/test/ui/raw_string.stderr]: The span should really only point at the first char and have a span label:

```

error: unterminated raw string
  --> $DIR/raw_string.rs:12:13
   |
12 |     let x = r##"lol"#;
   |             ^ unterminated raw string
   |
   = note: this raw string should be terminated with `"##`
```

**Comment by estebank** [comment]
Other than the nitpick, r=me

**Comment by GuillaumeGomez** [comment]
Updated.

File: src/libsyntax/parse/lexer/mod.rs [hunk 3]
```diff
@@ -269,6 +281,15 @@ impl<'a> StringReader<'a> {
         Self::push_escaped_char_for_msg(&mut m, c);
         self.fatal_span_(from_pos, to_pos, &m[..])
     }
+
+    fn struct_span_fatal(&self,
```
Comment (estebank) [inline src/libsyntax/parse/lexer/mod.rs]: Is this method necessary after the last change?

**Review by estebank — APPROVED** [review]
Lgtm, r=me after addressing the nit-pick.

File: src/libsyntax/parse/lexer/mod.rs (hunk 3 above)
Comment (GuillaumeGomez) [reply src/libsyntax/parse/lexer/mod.rs]: Still yes.

Comment (estebank) [reply src/libsyntax/parse/lexer/mod.rs]: Isn't it being used only in `fail_unterminated_raw_string` above? The span for the error and the span_label could be made only once (`self.mk_span(pos, pos)`) and the message and label literal could be the same variable.

**Pull request merged**